File: app/src/main/java/com/keepereye/app/obstacle/DetectedObstacle.kt
package com.keepereye.app.obstacle

import android.graphics.Rect

enum class ObstaclePosition {
    LEFT, CENTER, RIGHT
}

enum class ObstacleProximity(val label: String) {
    NEAR("muy cerca"),
    MEDIUM("a media distancia"),
    FAR("lejos")
}

data class DetectedObstacle(
    val label: String,
    val boundingBox: Rect,
    val position: ObstaclePosition,
    val proximity: ObstacleProximity,
    val confidence: Float,
    val trackingId: Int?
) {
    fun buildAlertMessage(): String {
        val posLabel = when (position) {
            ObstaclePosition.LEFT -> "a la izquierda"
            ObstaclePosition.CENTER -> "al frente"
            ObstaclePosition.RIGHT -> "a la derecha"
        }
        return when (proximity) {
            ObstacleProximity.NEAR -> "$label $posLabel, muy cerca"
            ObstacleProximity.MEDIUM -> "$label $posLabel"
            ObstacleProximity.FAR -> "$label $posLabel, lejos"
        }
    }

    companion object {
        private val LABEL_MAP = mapOf(
            "Fashion good" to "Persona u objeto",
            "Home good" to "Mueble",
            "Food" to "Objeto",
            "Place" to "Estructura",
            "Plant" to "Planta"
        )

        fun translateLabel(englishLabel: String): String {
            return LABEL_MAP[englishLabel] ?: "Obstáculo"
        }
    }
}

File: lib/models/detected_obstacle.dart
import 'dart:ui';

enum ObstaclePosition { left, center, right }

enum ObstacleProximity {
  near('muy cerca'),
  medium('a media distancia'),
  far('lejos');

  final String label;
  const ObstacleProximity(this.label);
}

class DetectedObstacle {
  final String label;
  final Rect boundingBox;
  final ObstaclePosition position;
  final ObstacleProximity proximity;
  final double confidence;
  final int? trackingId;

  const DetectedObstacle({
    required this.label,
    required this.boundingBox,
    required this.position,
    required this.proximity,
    required this.confidence,
    this.trackingId,
  });

  String buildAlertMessage() {
    final posLabel = switch (position) {
      ObstaclePosition.left => 'a la izquierda',
      ObstaclePosition.center => 'al frente',
      ObstaclePosition.right => 'a la derecha',
    };
    return switch (proximity) {
      ObstacleProximity.near => '$label $posLabel, muy cerca',
      ObstacleProximity.medium => '$label $posLabel',
      ObstacleProximity.far => '$label $posLabel, lejos',
    };
  }
}
